content-services/content-sources-backend
# Fixes 2443: API support for list / search package groups

## Summary

This adds:

- `repositories/{uuid}/package_groups` endpoint for listing, returns ID, Name, Description, and Package List metadata
- `package_groups/names` endpoint for searching, returns Name, Description, and Package List of all groups found across the repositories requested
- database support to store package group metadata
- groups are de-duplicated based on combination of name/ID/package list
- search API returns full list of packages that are associated with the search term and repositories

## Testing steps

Endpoint: `$BASE_URL/package_groups/names`

Example request: `{
  "urls": [
    "https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/"
  ],
  "search": "xfce"
}`

- add a repository that has a comps.xml file (e.g. https://dl.fedoraproject.org/pub/epel/8/Everything/x86_64/) and make a request to `repositories/{uuid}/package_groups`, should list package groups associated with that repo
- make a request to `package_groups/names` with a valid repo URL or repo UUID and a search term, should return the package groups with that search term in the name
** for example, the epel 8 and epel 9 repos have some package groups (one is `xfce`) with different package lists. if you add both of these repos and include both repos in the request, the search API should return the full list of packages from both repos like this:
`[
    {
        "package_group_name": "Xfce",
        "description": "A lightweight desktop environment that works well on low end machines.",
        "package_list": [
            "xfce4-screensaver",
            "tumbler",
            "NetworkManager-gnome",
            "openssh-askpass",
            "gdm",
            "xfce4-terminal",
            "thunar-volman",
            "xfce4-panel",
            "xfce4-appfinder",
            "network-manager-applet",
            "xfwm4",
            "xfce4-settings",
            "xfce4-pulseaudio-plugin",
            "pinentry-gtk",
            "mousepad",
            "xfdesktop",
            "xfconf",
            "Thunar",
            "xfce-polkit",
            "thunar-archive-plugin",
            "xfce4-session",
            "pinentry-gnome3",
            "xfce4-power-manager"
        ]
    }
]`

## Checklist
